### PR TITLE
Fix more problems with feed overrides

### PR DESCRIPTION
**Pull request by alexperovich**
### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/model/SetupTargetFeedConfigV3.cs [hunk 1]
```diff
@@ -172,7 +170,7 @@ private IEnumerable<TargetFeedConfig> Feeds()
                         type,
                         sasUri ?? feed,
                         feedType,
-                        key,
+                        sasUri == null ? key : null,
```
Comment (adiaaida) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/model/SetupTargetFeedConfigV3.cs:173]: if it's null, use the key, otherwise use null?

Comment (alexperovich) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/model/SetupTargetFeedConfigV3.cs:173]: yea, if we have a sasUri, it should prefer using that, so don't use the key if we have a sasUri.

Comment (adiaaida) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/model/SetupTargetFeedConfigV3.cs:173]: Oh I see. I was definitely misreading this

**Review by adiaaida — APPROVED** [review]
:shipit:

**Pull request merged**